File: AL/nadongbin/01-greedy/numberCardGame/numberCardGame.ts
function numberCardGame(cards: number[][]) {
  let max = 0
  for (let i = 0; i < cards.length; i++) {
    const min = Math.min(...cards[i])
    max = Math.max(max, min)
  }
  return max
}

const cards = [
  [3, 1, 2],
  [4, 1, 4],
  [2, 2, 2],
]

console.log(numberCardGame(cards))

File: AL/nadongbin/01-greedy/numberCardGame/numberCardGame.go
package main

import "fmt"

func main() {
	cards := [][]int{
		{3, 1, 2},
		{4, 1, 4},
		{2, 2, 2},
	}
	fmt.Println(numberCardGame(cards))
}

func numberCardGame(cards [][]int) int {
	rowMin := 100001
	max := 0
	for _, rows := range cards {
		for _, card := range rows {
			if rowMin > card {
				rowMin = card
			}
		}
		if rowMin > max {
			max = rowMin
		}
		rowMin = 100001
	}
	return max
}
